### PR TITLE
AP-876 CCMS Asset attributes

### PR DESCRIPTION
**Pull request by stephenrichards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-876)

Most of the attributes in the ticket had already been done.

Ones which were changes were:
- APPLICATION_CASE_REF
- GB_INPUT_B_2WP2_1A

Also changed the `AttributeValueGenerator` initializer to take CCMS::Submission record as a parameter rather than the `LegalAidApplication` - this is because we need the `case_ccms_reference` number from that record, and there could conceivably be more than one per application, so this ensures we get the right one.

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/services/ccms/case_add_requestor_xml_blocks_spec.rb [hunk 1]
```diff
@@ -103,6 +105,36 @@ module CCMS # rubocop:disable Metrics/ModuleLength
...
+
+        context 'is a beneficiary' do
+          it 'inserts true into the attribute block' do
+            block = XmlExtractor.call(xml, :global_means, 'GB_INPUT_B_2WP2_1A')
+            expect(block).to have_boolean_response true
+          end
+        end
+      end
+
+      context 'GB_INPUT_B_3WP2_1A - applicant has financial interest in his main home' do
```
Comment (mpw5) [inline spec/services/ccms/case_add_requestor_xml_blocks_spec.rb]: Should there be tests here?

File: spec/services/ccms/case_add_requestor_xml_blocks_spec.rb.orig [hunk 2]
```diff
@@ -0,0 +1,1451 @@
+require 'rails_helper'
```
Comment (mpw5) [inline spec/services/ccms/case_add_requestor_xml_blocks_spec.rb.orig]: I don't think this file is needed.

**Comment by scruti** [comment]
I am confused about the couple "rebase" commits on this branch.
It looks like a "merge master" but with rebase title 😅, so not sure if I understood what is happening there.

Are we separately committing the rebase conflicts resolution? In that case shouldn't those conflict resolutions just amend the particular commit conflicting against master? 

I may be missing or may be a temporal thing pre-squashing & merging, just wanted to understand how those commits pop out :)

File: spec/services/ccms/case_add_requestor_xml_blocks_spec.rb (hunk 1 above)
Comment (stephenrichards) [reply spec/services/ccms/case_add_requestor_xml_blocks_spec.rb]: Erm NO!

Comment (stephenrichards) [reply spec/services/ccms/case_add_requestor_xml_blocks_spec.rb]: Erm.  YES

**Comment by stephenrichards** [comment]
> I am confused about the couple "rebase" commits on this branch.
> It looks like a "merge master" but with rebase title 😅, so not sure if I understood what is happening there.
> 
> Are we separately committing the rebase conflicts resolution? In that case shouldn't those conflict resolutions just amend the particular commit conflicting against master?
> 
> I may be missing or may be a temporal thing pre-squashing & merging, just wanted to understand how those commits did show up :)

I've no idea how this happened Marc.  I never merge master, but could it have been a rebase which required some changes?

**Pull request merged**